test/controllers/journeys/income/AnyOtherIncomeControllerSpec.scala

```diff
@@ -17,40 +17,52 @@
 package controllers.journeys.income
 
 import base.SpecBase
-import controllers.journeys.income.routes.AnyOtherIncomeController
+import controllers.journeys.income.routes.{
+  AnyOtherIncomeController,
+  OtherIncomeAmountController,
+  TradingAllowanceController,
+  TurnoverNotTaxableController
+}
 import controllers.standard.routes.JourneyRecoveryController
 import forms.income.AnyOtherIncomeFormProvider
 import models.{CheckMode, NormalMode, UserAnswers}
-import navigation.{FakeNavigator, Navigator}
-import org.mockito.ArgumentMatchers.any
+import navigation.{FakeIncomeNavigator, IncomeNavigator}
+import org.mockito.ArgumentMatchers.{any, eq => meq}
 import org.mockito.Mockito.when
 import org.scalatestplus.mockito.MockitoSugar
 import pages.income.AnyOtherIncomePage
 import play.api.data.Form
 import play.api.i18n.I18nSupport.ResultWithMessagesApi
 import play.api.i18n.MessagesApi
 import play.api.inject.bind
-import play.api.mvc.Call
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import repositories.SessionRepository
+import services.SelfEmploymentService
 import views.html.journeys.income.AnyOtherIncomeView
 
 import scala.concurrent.Future
 
 class AnyOtherIncomeControllerSpec extends SpecBase with MockitoSugar {
 
-  def onwardRoute = Call("GET", "/foo")
+  val formProvider           = new AnyOtherIncomeFormProvider()
+  val businessId             = "SJPR05893938418"
+  val otherIncomeAmountCall  = OtherIncomeAmountController.onPageLoad(taxYear, businessId, NormalMode)
+  val turnoverNotTaxableCall = TurnoverNotTaxableController.onPageLoad(taxYear, businessId, NormalMode)
+  val tradingAllowanceCall   = TradingAllowanceController.onPageLoad(taxYear, businessId, NormalMode)
+
+  val mockService: SelfEmploymentService       = mock[SelfEmploymentService]
+  val mockSessionRepository: SessionRepository = mock[SessionRepository]
+
+  val answerIsTrueOnwardRoute = otherIncomeAmountCall
 
-  val formProvider       = new AnyOtherIncomeFormProvider()
-  val formWithIndividual = formProvider("individual")
-  val formWithAgent      = formProvider("agent")
+  def answerIsFalseOnwardRoute(isAccrual: Boolean) = if (isAccrual) turnoverNotTaxableCall else tradingAllowanceCall
```

Comment:
Bit more readable to pass in the accounting type to this method and pattern match on its value, to avoid passing in boolean values. 

I think the method name could be improved, e.g. `redirectCallFromNoResponse` or wtte, so this could all become:

`def redirectCallFromNoResponse(accountingType: String) = accountingType match {
       case "ACCRUAL" => turnoverNotTaxableCall
       case "CASH"        => tradingAllowanceCall
}`